tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_models/test_dataset_io_configuration_model.py

```diff
@@ -89,8 +90,12 @@ def get_data_io_kwargs(self):
     data = np.array(["test", "string", "abc"], dtype=object)
     nwbfile.add_trial_column(name="test", description="test column with object dtype but all strings", data=data)
     neurodata_object = nwbfile.trials.columns[2]
+    manager = get_manager()
```

Comment:
Do we still need this here after making the builder optional?

Reply:
Not strictly speaking, so I made one version of the test with a builder and one without.

Reply:
How are you thinking about it? 
If you are testing consistency between passing builder and not builder shouldn't they be the same test to see that the output is the same?

To be honest I was just thinking in not having a test for the builder at this level. This behavior does not dependent on it (it is only relevant for compound data types) but maybe I am missing something, how are you thinking?

Reply:
I added it for completeness (testing all the paths that the code could travel through), but honestly I agree that not having a test for builder behavior at this level is fine since it will be tested at the `get_default_dataset_io_configurations` level and at the `get_full_data_shape` level.